tutorials/video_classification_example/models.py

```diff
@@ -0,0 +1,145 @@
+            lr (float, optional): The learning rate for the Adam optimizer. Defaults to 2e-4.
+            freeze_backbone (bool, optional): Whether to freeze the backbone or leave it trainable. Defaults to True.
+            pretrained (bool, optional): Use the pretrained model from torchhub. When False, we initialize the
+            slow_r50 model from scratch. Defaults to True.
+
+        All extra kwargs will be available via self.hparams.<name-of-arg>. These will also be saved as
+        TensorBoard Hparams.
+        """
+        super().__init__()
+
+        # Saves all kwargs to self.hparams. Use references to self.hparams.<var-name>, not the init args themselves.
+        self.save_hyperparameters()
+
+        # Build the model in separate function so its easier to override.
+        self.model = self._build_model()
+
+        # Metrics we will keep track of.
+        self.loss_fn = nn.CrossEntropyLoss()
+        self.train_acc = pl.metrics.Accuracy()
+        self.val_acc = pl.metrics.Accuracy()
+        self.accuracy = {"train": self.train_acc, "val": self.val_acc}
+
+    def _build_model(self):
+        return create_resnet(model_num_class=self.hparams.num_classes)
+
+    def on_train_epoch_start(self):
+        """
+        For distributed training we need to set the datasets video sampler epoch so
+        that shuffling is done correctly
+        """
+        epoch = self.trainer.current_epoch
+        if self.trainer.use_ddp:
+            self.trainer.datamodule.train_dataset.dataset.video_sampler.set_epoch(epoch)
+
+    def forward(self, x: torch.Tensor):
+        """
+        Forward defines the prediction/inference actions.
+        """
+        return self.model(x)
+
+    def shared_step(self, batch, mode: str):
+        """This shared step handles both the training and validation steps to avoid
+        re-writing the same code more than once. The given `mode` will change the name
+        of the logged metrics.
+
+        PyTorchVideo batches are dictionaries containing each modality or metadata of
+        the batch collated video clips. Kinetics contains the following notable keys:
+           {
+               'video': <video_tensor>,
+               'label': <action_label>,
+           }
+
+        - "video" is a Tensor of shape (batch, channels, time, height, Width)
+        - "label" is a Tensor of shape (batch, 1)
+
+        The PyTorchVideo models and transforms expect the same input shapes and
+        dictionary structure making this function just a matter of unwrapping the dict and
+        feeding it through the model/loss.
+
+        Args:
+            batch (dict): PyTorchVideo batch dictionary containing a single batch of data.
+            mode (str): The type of step. Can be 'train', 'val', or 'test'.
+
+        Returns:
+            torch.Tensor: The loss for a single batch step.
+        """
+
+        # Pass video tensor through model to get outputs.
```

Comment:
Can you remove all these inline comments, way to verbose imo

Reply:
Done :rocket: